pulsar/internal/utils.go

```diff
@@ -40,7 +40,7 @@ func TimestampMillis(t time.Time) uint64 {
 // GetAndAdd perform atomic read and update
 func GetAndAdd(n *uint64, diff uint64) uint64 {
 	for {
-		v := *n
+		v := atomic.LoadUint64(n)
 		if atomic.CompareAndSwapUint64(n, v, v+diff) {
```

Comment:
I'm not familiar enough with the pulsar codebase to understand what this is trying to achieve, but the usual way to perform an atomic get-and-add on a uint64 in golang is to use `atomic.AddUint64()`. Also, is this change related to the issue at hand, or is it an unrelated change?

Reply:
This change is to fix a race issue in the CI discovered by the new test case `TestSendConcurrently`.
The difference between them is that atomic.AddUint64() returns the new value but `GetAndAdd ` here returns the old value.
Actually I was trying to [use atomic.AddUint64()](https://github.com/apache/pulsar-client-go/pull/1164/commits/51a62c212fa1d860b162c076606484f47e69e951), but if failed in the CI because [there are other cases](https://github.com/apache/pulsar-client-go/actions/runs/7736056342/job/21092758357#step:5:9550) that need the old values.
And yes, this is to fix another regression bug introduced in 0.12.0.